components/mpas-ocean/src/mode_init/mpas_ocn_init_global_ocean.F

```diff
@@ -1205,29 +1019,23 @@ subroutine ocn_init_setup_global_ocean_interpolate_land_ice_topography(domain, i
                'indicates that this field is missing in the initial condition file (e.g. ' // &
                'because it is meant for an older E3SM version).', MPAS_LOG_CRIT)
           endif
-          ssh(:) = 0.0_RKIND
           landIceFraction(:) = 0.0_RKIND
-          modifyLandIcePressureMask(:) = 0
+          landIceDraft(:) = 0.0_RKIND
           landIcePressure(:) = 0.0_RKIND
+          sshAdjustmentMask(:) = 0
           do iCell = 1, nCells
-
-             if (landIceMask(iCell) == 1) then
-                landIceFraction(iCell) = landIceFracObserved(iCell)
-             end if
-
-             ! this implicitly assumes that when a cell is considered floating, the
-             ! full landIceFracObserved is floating
-             if (landIceFloatingMask(iCell) == 1) then
-                landIceFloatingFraction(iCell) = landIceFloatingFracObserved(iCell)
-             end if
-
              ! nothing to do here if the cell is land
              if (maxLevelCell(iCell) <= 0) cycle
 
-             ! we compute the SSH first and find out the land-ice pressure
-             ssh(iCell) = min(0.0_RKIND, landIceDraftObserved(iCell))
-             if (ssh(iCell) < 0.0_RKIND) then
-                modifyLandIcePressureMask(iCell) = 1
+             ! don't mask these because we may want to include melt that bleeds
+             ! into the open ocean
+             landIceFraction(iCell) = landIceFracObserved(iCell)
+             landIceFloatingFraction(iCell) = landIceFloatingFracObserved(iCell)
```

Comment:
In case anyone has been reviewing this already, I just added the commit to change this.  I think we don't want to hard-code this masking here because we want some flexibility about how to allow fluxes to bleed into cells that are partly open ocean and partly ice.  This will be important for conservation when coupling with MALI (without requiring rerouting to cells the ocean decides are under land ice, which I think we want to avoid).  I don't see an clear harm in at least supporting this possibility.